server/modules/doppler-client.js

```diff
@@ -0,0 +1,145 @@
+const querystring = require('querystring');
+const fetch = require('node-fetch');
+const shopify = require ('./shopify-extras');
+const baseUrl = 'https://restapi.fromdoppler.com';
+
+const getCustomerFieldValue = function(customer, fieldPath) {
+    let currentProperty = customer;
+    
+    fieldPath.split(".").forEach(propertyName => {
+        currentProperty = currentProperty[propertyName];
+    });
+    
+    return currentProperty;
+}
+
+class Doppler {
+    constructor(accountName, apiKey) {
+        this.accountName = querystring.escape(accountName);
+        this.apiKey = querystring.escape(apiKey);
+    }
+
+    async sendRequestAsync(url, fetchOptions) {
+        const response = await fetch(url, fetchOptions);
+        const responseBody = await response.json();
+
+        if (response.status >= 400) {
+            const msg = responseBody && (responseBody.title || responseBody.detail)
+                ? `${responseBody.title ? `${responseBody.title}: ` : ''}${responseBody.detail ? responseBody.detail : ''}`
+                : 'Unexpected error';
+    
+            throw new Error(msg);
+        }
+
+        return responseBody
+    }
+
+    async AreCredentialsValidAsync() {
+        const url = `${baseUrl}/accounts/${this.accountName}?api_key=${this.apiKey}`;
+        
+        try {
+            await this.sendRequestAsync(url);
+            return true;
+        } catch (error) {
+            console.warn(`Error validating credentials: ${JSON.stringify(error)}`);
+            return false;
+        }
+    }
+
+    async getListsAsync() {
+        // TODO: what if there are more than 200 lists?
+        const url = `${baseUrl}/accounts/${this.accountName}/lists?page=1&per_page=200&state=active&api_key=${this.apiKey}`;
```

Comment:
If there are more than 200 lists you will know it based on the result. What do you think about simply show a warning?

For example _"There are more than 200 lists. Some lists are not visible."_

Reply:
OK, sounds good. Will return the total number of lists in the response.